### ISSUE TITLE
Throw when a promise is returned from an `it` mocha test during an e2e / integration test

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/cypress#464]
Related to #435 and #463.

This is almost the same issue except:
- we will throw a slightly more explicit error message
- we will only throw if `cy` was ever invoked

Example:

```js
it("should throw", function(){
  // this should explode
  return new Promise((resolve, reject) => {
    cy.request("/foo").then(resolve)
  })
})

it("does not throw", function(){
  // if cy is not invoked then we're good to go
  return new Promise.delay(1000)
  .then(function(){
     // nothing wrong with this since
     // we never invoked cy
   })
})
```

**Comment by brian-mann** [comment]
Instead of throwing an error we will print a warning to the console.

![screen shot 2017-08-24 at 4 13 35 pm](https://user-images.githubusercontent.com/1268976/29686635-be2af090-88e7-11e7-837e-14b218eb91fb.png)

**Comment by brian-mann** [comment]
The code for this is done, but this has yet to be released. We'll update the issue and reference the changelog when it's released.

**Comment by brian-mann** [comment]
Fixed in [`0.20.0`](https://docs.cypress.io/guides/references/changelog.html#0-20-0)
